pkg/registry/service/ipallocator/allocator.go

```diff
@@ -107,7 +107,7 @@ func (r *Range) Free() int {
 func (r *Range) Allocate(ip net.IP) error {
 	ok, offset := r.contains(ip)
 	if !ok {
-		return ErrNotInRange
+		return errors.New(fmt.Sprintf("%v %v", r, ip))
```

Comment:
@jayunit100  In general, when you see an exported error being returned, that's a strong hint it's a contract you shouldn't replace with an arbitrary error